components/brave_ads/browser/search_result_ad/search_result_ad_util.h

```diff
@@ -0,0 +1,18 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_ADS_BROWSER_SEARCH_RESULT_AD_SEARCH_RESULT_AD_UTIL_H_
+#define BRAVE_COMPONENTS_BRAVE_ADS_BROWSER_SEARCH_RESULT_AD_SEARCH_RESULT_AD_UTIL_H_
+
+#include "brave/components/brave_ads/browser/search_result_ad/search_result_ad_aliases.h"
+#include "third_party/blink/public/mojom/document_metadata/document_metadata.mojom.h"
+
+namespace brave_ads {
+
+SearchResultAdsList ParseWebPageEntities(blink::mojom::WebPagePtr web_page);
```

Comment:
why is this in a separate util file? Seems like this should be either a private or anonymous method in SearchResultAdHandler

Reply:
there are several goals of this:
- logic separation
- file size reduce
- ability to test parsing logic separately

Reply:
moved to anonymous method